libbeat/dashboards/dashboards.go

```diff
@@ -50,6 +50,18 @@ func ImportDashboards(
 		return err
 	}
 
+	if esConfig.Enabled() {
+		username, _ := esConfig.String("username", -1)
+		password, _ := esConfig.String("password", -1)
+
+		if !kibanaConfig.HasField("username") && username != "" {
+			kibanaConfig.SetString("username", -1, username)
+		}
+		if !kibanaConfig.HasField("password") && password != "" {
+			kibanaConfig.SetString("password", -1, password)
+		}
+	}
```

Comment:
I know this is some old code re-introduced. But in general: We should not use SetX on globally shared configuration objects. Instead of passing through `*common.Config`, as is done here, we should strive for concrete types and unpack as early as possible.

Reply:
@urso Would it be ok for you to get this in as is for FF and we open an issue to clean this up?

Reply:
@urso Thanks for pointing this out. I will push a new PR later to address your comment and add some tests as well.